### ISSUE TITLE
Stream power non-integer n bug

### ISSUE THREAD

**Issue by gregtucker** [landlab/landlab#431]
In stream_power/cfuncs.pyx:

                z_diff = next_z - z[dst_id]
                f = alpha[src_id] * pow(z_diff, n - 1.)

if lakes/depressions are present, z_diff can be < 0. If n is not an integer, this results in an undefined operation: fractional power of a negative number. The proper behavior should be, I think, to set f = 0 when z_diff < 0.

**Comment by SiccarPoint** [comment]
Manu Schmid has been fighting this guy, and I'm approaching a solution. Assigning myself.

**Comment by SiccarPoint** [comment]
An alternative is simply to break the iterative loop if slopes <= 0 are seen, which has the advantage of saving a bit of calculation...

**Comment by gregtucker** [comment]
@SiccarPoint take a look at PR 433

**Comment by ManuSchmi88** [comment]
Hey, sorry for chipping in to this. I meddled with this issue for quite while now. Something that I tried yesterday which seems to work for me (I haven't had a look at the file SiccarPoint sent me per slack) was to do this: 
if z_diff < 0:
      f = 0
else:
     f = alpha[src_id] * pow(z_diff, n - 1.)

I guess this is was greg tucker suggested. I am still running tests if this works for all my cases but at least a few which I had problems with are now running.
